server/schedulers/hot_region.go

```diff
@@ -196,56 +192,47 @@ func (h *hotScheduler) dispatch(typ rwType, cluster opt.Cluster) []*operator.Ope
 func (h *hotScheduler) prepareForBalance(cluster opt.Cluster) {
 	h.summaryPendingInfluence()
 
-	h.readScores = h.analyzeStoreLoad(cluster, read)
-	h.writeScores = h.analyzeStoreLoad(cluster, write)
-
 	storesStat := cluster.GetStoresStats()
 
+	minHotDegree := cluster.GetHotRegionCacheHitsThreshold()
 	{ // update read statistics
 		regionRead := cluster.RegionReadStats()
 		storeRead := storesStat.GetStoresBytesReadStat()
 
-		asLeader := calcScore(regionRead, storeRead, cluster, core.LeaderKind)
-		h.stats.readStatAsLeader = h.calcPendingInfluence(asLeader, h.readPendingSum)
+		h.stLoadInfos.ReadLeaders = summaryStoresLoad(h.GetName(),
+			storeRead,
+			h.readPendingSum,
+			regionRead,
+			minHotDegree,
+			read, core.LeaderKind)
 	}
 
 	{ // update write statistics
 		regionWrite := cluster.RegionWriteStats()
 		storeWrite := storesStat.GetStoresBytesWriteStat()
 
-		asLeader := calcScore(regionWrite, storeWrite, cluster, core.LeaderKind)
-		h.stats.writeStatAsLeader = h.calcPendingInfluence(asLeader, h.writePendingSum)
-
-		asPeer := calcScore(regionWrite, storeWrite, cluster, core.RegionKind)
-		h.stats.writeStatAsPeer = h.calcPendingInfluence(asPeer, h.writePendingSum)
-	}
-}
+		h.stLoadInfos.WriteLeaders = summaryStoresLoad(
+			h.GetName(),
+			storeWrite,
+			map[uint64]Influence{},
+			regionWrite,
+			minHotDegree,
+			write, core.LeaderKind)
 
-func (h *hotScheduler) calcPendingInfluence(storeStat statistics.StoreHotPeersStat, pending map[uint64]Influence) statistics.StoreHotPeersStat {
-	for id, stat := range storeStat {
-		stat.FutureBytesRate += pending[id].ByteRate
+		h.stLoadInfos.WritePeers = summaryStoresLoad(
+			h.GetName(),
+			storeWrite,
+			h.writePendingSum,
+			regionWrite,
+			minHotDegree,
+			write, core.RegionKind)
 	}
-	return storeStat
 }
 
-func filterUnhealthyStore(cluster opt.Cluster, storeStatsMap map[uint64]float64) {
-	stores := cluster.GetStores()
-	for _, store := range stores {
-		if store.IsTombstone() ||
-			store.DownTime() > cluster.GetMaxStoreDownTime() {
-			delete(storeStatsMap, store.GetID())
-		}
-	}
-}
-
-func (h *hotScheduler) updateStatsByPendingOpInfo(storeStatsMap map[uint64]float64, balanceType rwType) {
-	for storeID := range storeStatsMap {
-		if balanceType == read {
-			storeStatsMap[storeID] += h.readPendingSum[storeID].ByteRate
-		} else {
-			storeStatsMap[storeID] += h.writePendingSum[storeID].ByteRate
-		}
-	}
+func (h *hotScheduler) summaryPendingInfluence() {
+	h.readPendingSum = summaryPendingInfluence(h.readPendings, calcPendingWeight)
+	h.writePendingSum = summaryPendingInfluence(h.writePendings, calcPendingWeight)
```

Comment:
Is it necessary to make `calcPendingWeight` as a parameter?

Reply:
This is not new code, it is moved here from line 881. I designed it to make `summaryPendingInfluence` more flexible. We may use some other function to calculate the weight.